src/App/Expression.cpp

```diff
@@ -2007,117 +2110,175 @@ Py::Object FunctionExpression::evaluate(const Expression *expr, int f, const std
     if (f > AGGREGATES)
         return evalAggregate(expr, f, args);
 
-    if(f == LIST) {
-        if(args.size() == 1 && args[0]->isDerivedFrom(RangeExpression::getClassTypeId()))
+    switch (f) {
```

Comment:
I'm not sold on changing this to a `switch` that only handles a subset of the cases, then stops, inserts a conditional, then goes back into a switch statement: that's pretty non-idiomatic. You're going to get static analyzers complaining about missing cases, etc.

Reply:
Isn't it more idiomatic for the `f` argument to be consistently tested? I don't see the separation between this switch/case block and the next. Having no default, although an issue in other languages isn't an issue in C++. I also don't see any lint for it reported either. This is also more scalable: as more items are added, the compiler should be using a jump table for O(1) access instead of a list of if/else which will be O(n).

Reply:
Given that the C++ Core Guidelines admonish us to use switch whenever possible, I'll merge this as-is.